### DISCUSSION TITLE
Add Default: Random for Inputs

### DISCUSSION THREAD

**Discussion by TXAE** [hneemann/Digital#920] [Ideas]
Add option to set the default value of an input to random (e.g. the output of a gate/driver that's on and is not-connected on input side).

**Comment by hneemann** [comment]
Why not connecting the "not connected" component?

**Reply by TXAE** [reply]
Not connected symbol not allowed directly on input.
My way to randomize an input is this and it's a hassle compared to if the input had a setting "random" for default.
Is there a better way to randomize (not float/High-Z) an input?
[Random_Input.zip](https://github.com/hneemann/Digital/files/8055443/Random_Input.zip)

**Reply by hneemann** [reply]
But if a high-z value is read it's value is random. Thus high-z and "random" is the same.
And all ot this is possible:

![z](https://user-images.githubusercontent.com/7190563/153753695-90e9e845-a7e2-4c19-b28a-4c58a041f46c.png)

**Reply by TXAE** [reply]
Yeah it's a visual thing. I would like to have the option for an input to show a random number instead of High-Z (even though they're read as the same), without the workaround with a driver and not connected (as I said, not connected not allowed directly on an input. You show this with an output, where it is possible). 
Minor :)

**Reply by hneemann** [reply]
To be honest, I don't get it. 
Imagine the circuit like a real circuit. What happens in reality when a wire is not connected to anything and a digital input stage is connected to this wire depends on the technology of the circuit. With CMOS, some random value is read as the potential of this wire drifts towards the center potential. This can be interpreted as a "high" or a "low", but it can also destroy the input stage, because both transistors of the input inverter can become conductive. In the simulator it doesn't matter to leave an input open because a random value is generated to mimic the harmless behavior, in reality this is dangerous. So it is strange to use this effect intentionally to display random numbers, especially since in reality of course random numbers are not really generated, the value is rather undefined. It can also be that the value is always "high" or "low".
